modules/mobile/app/controllers/mobile/v0/appointments_controller.rb

```diff
@@ -81,19 +84,26 @@ def fetch_cached_or_service(validated_params)
 
         appointments.filter! { |appt| appt.is_pending == false } unless validated_params[:included]&.include?('pending')
         appointments.reverse! if validated_params[:reverse_sort]
+
         appointments
       end
 
       def paginate(appointments, validated_params)
         appointments = appointments.filter do |appointment|
           appointment.start_date_utc.between?(
-            validated_params[:start_date], validated_params[:end_date]
+            validated_params[:start_date].beginning_of_day, validated_params[:end_date].end_of_day
           )
         end
         url = request.base_url + request.path
         Mobile::PaginationHelper.paginate(list: appointments, validated_params: validated_params, url: url)
       end
 
+      def include_pending_in_links(links)
```

Comment:
I believe this should be sufficient because there should always be other query params. If I'm wrong, I can make this more robust with the URI library.